src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,100 @@
+            attendanceAssignment.setStatus(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS);
+            return attendanceAssignment;
+        }
+
+        // Looks for the linked assignment in canvas
+        Optional<Assignment> assignmentOptional = lookForAssignmentInCanvas(courseId, attendanceAssignment, canvasApiWrapperService, oauthToken);
+
+        // We will create a new assignment for the instructor based on what's left in the database
+        if (!assignmentOptional.isPresent()) {
+            LOG.info("No Canvas assignment linked to section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId());
+            attendanceAssignment.setStatus(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS);
+        }
+
+        return attendanceAssignment;
+    }
+
+    public AttendanceAssignment validateCanvasAssignment(AttendanceAssignment assignmentConfigurationFormSetup, Long courseId, AttendanceAssignment attendanceAssignment,
+                                         CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) throws AttendanceAssignmentException{
+
+        //Look for changes between the course configuration in the form and in the DB
+        if(!isAssignmentConfigurationSaved(assignmentConfigurationFormSetup, attendanceAssignment)) {
+            LOG.info("Configuration form is different than saved assignment configuration for section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId());
+            throw new AttendanceAssignmentException(Error.NOT_SAVED);
+
+        }
+
+        //Look for the canvas assignment in canvas
+        Optional<Assignment> assignmentOptional = lookForAssignmentInCanvas(courseId,attendanceAssignment,canvasApiWrapperService,oauthToken);
+
+        //Looks for discrepancy between the assignment in canvas and in database
+        if (assignmentOptional.get().getPointsPossible().doubleValue() != assignmentConfigurationFormSetup.getAssignmentPoints().doubleValue()) {
+            LOG.debug("Discrepancy between Canvas and DB assignment. Point value of Canvas assignment is: " + assignmentOptional.get().getPointsPossible() +
+                      " and point value of Database assignment is :" + attendanceAssignment.getAssignmentPoints());
+            attendanceAssignment.setStatus(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY);
+        }
+        return attendanceAssignment;
+    }
+
+    /**
+     * Returns true if configuration in the form and in the db is the same
+     */
+    private boolean isAssignmentConfigurationSaved(AttendanceAssignment assignmentConfigurationFromSetup, AttendanceAssignment attendanceAssignmentSaved) {
+        return assignmentConfigurationFromSetup.getAssignmentPoints().doubleValue() == attendanceAssignmentSaved.getAssignmentPoints().doubleValue()
+                && assignmentConfigurationFromSetup.getExcusedPoints().doubleValue() == attendanceAssignmentSaved.getExcusedPoints().doubleValue()
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1244)